mitmproxy/protocol/base.py

```diff
@@ -151,11 +159,16 @@ def disconnect(self):
         """
         self.log("serverdisconnect", "debug", [repr(self.server_conn.address)])
         address = self.server_conn.address
-        source_address = self.server_conn.source_address
         self.server_conn.finish()
         self.server_conn.close()
         self.channel.tell("serverdisconnect", self.server_conn)
-        self.server_conn = models.ServerConnection(address, (source_address.host, 0))
+
+        if self.config.options.spoof_source_address:
```

Comment:
True, but isn't `self.server_conn.source_address` guaranteed to be `self.ctx.client_conn.address.host` already based on the changes in the constructor?